### ISSUE TITLE
chunkserver hdd_random_chunk crash

### ISSUE THREAD

**Issue by nickb937** [moosefs/moosefs#203]
Commit dd2581433b997af1470bb1771d3d910d946b3e46 introduces a crash in hdd_random_chunk().
The problem is that `c->damaged` is being used before `c` is assigned.

```
(gdb) bt
#0  0x0000555555577188 in hdd_random_chunk (f=f@entry=0x55555d5ed7f0) at hddspacemgr.c:4642
#1  0x000055555557787b in hdd_int_move (fsrc=fsrc@entry=0x55555d5ed7f0, fdst=fdst@entry=0x55555d5d0ca0) at hddspacemgr.c:4718
#2  0x00005555555790ed in hdd_rebalance_thread (arg=<optimized out>) at hddspacemgr.c:5343
#3  0x00007ffff7bf2a9d in start_thread () from /usr/lib/libpthread.so.0
#4  0x00007ffff7b22b23 in clone () from /usr/lib/libc.so.6
(gdb) 

Thread 7 "mfschunkserver" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff6d84700 (LWP 12716)]
0x0000555555577188 in hdd_random_chunk (f=f@entry=0x55555d5ed7f0) at hddspacemgr.c:4642
4642				if (f->chunktab[pos]->state==CH_AVAIL && c->damaged==0) {
(gdb) print f
$1 = (folder *) 0x55555d5ed7f0
(gdb) print pos
$2 = 0
(gdb) print f->chunktab[pos]
$3 = (struct chunk *) 0x7fffe8008c00
(gdb) print f->chunktab[pos]->state
$4 = 0 '\000'
(gdb) print c
$5 = (chunk *) 0x0
(gdb) 
```

**Comment by oxide94** [comment]
Hi,
Thanks for your report, @acid-maker discovered it today independently and it is already fixed. A relevant commit will be pushed soon.

Best,
Piotr

**Comment by oxide94** [comment]
Related to #192.

**Comment by acid-maker** [comment]
fixed by 778a323807c3b8dc587522a273fc87331c824e95
